types/types.go

```diff
@@ -23,67 +24,177 @@ import (
 	"github.com/prometheus/common/model"
 )
 
+type State uint8
+
+const (
+	Unprocessed State = iota
+	Active
+	Silenced
+	Inhibited
+)
+
+// AlertStatus stores the state and values associated with an Alert.
+type AlertStatus struct {
+	status State
+	values []string
+}
+
+// Save an allocation when serializing an empty string slice.
+var emptyVals = []string{}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (a *AlertStatus) MarshalJSON() ([]byte, error) {
+	values := a.values
+	if values == nil {
+		values = emptyVals
+	}
+	return json.Marshal(map[string]interface{}{
+		"status": a.status,
+		"values": values,
+	})
+}
+
+// Save an allocation when serializing the unknown response.
+const unknown = "unknown"
+
+// MarshalJSON implements the json.Marshaler interface.
+func (s State) MarshalJSON() ([]byte, error) {
+	status, found := statusMap[s]
+	if !found {
+		status = unknown
+	}
+
+	return json.Marshal(status)
+}
+
+var statusMap = map[State]string{
+	Unprocessed: "unprocessed",
+	Active:      "active",
+	Silenced:    "silenced",
+	Inhibited:   "inhibited",
```

Comment:
I think there's a difference between silenced and inhibited from the user perspective and Alertmanager should distinguish between these two states. Inhibition and silencing is used for different purposes. One is "oh it's silenced, someone is working on it" the other is "oh it's caused by some other issue" (btw. there's no way to ACK an alert, so silences might be used as ACKs).
If alert is inhibited than it's clear that it's suppressed by another alert, so when that other alert is resolved our inhibited alert will fire. Silenced alert on the other hand will never fire while the silence is active. If you just see "suppressed" you won't know what that really means and whenever this alert will fire if there's an action to take in the near future (it won't if it's silenced).